src/isArray.test.ts

```diff
@@ -2,6 +2,13 @@ import { isArray } from './isArray';
 import { typesDataProvider } from '../test/types_data_provider';
 
 describe('isArray', () => {
+  test('isArray: should infer ReadonlyArray<unknown> when given any', () => {
+    const data1 = {} as any
+    if (isArray(data1)) {
+      const unknownArray = [] as ReadonlyArray<unknown>;
+      expectTypeOf(data1).toEqualTypeOf(unknownArray);
```

Comment:
i tried using `assertType` instead of `expectTypeOf` but it wasn't reporting errors when it should.
This way it fails when the value is inferred as anything else other `ReadonlyArray<unknown>`